src/Microsoft.AspNet.Mvc.Core/MvcRouteHandler.cs

```diff
@@ -102,14 +108,23 @@ public VirtualPathData GetVirtualPath([NotNull] VirtualPathContext context)
             var invoker = invokerFactory.CreateInvoker(actionContext);
             if (invoker == null)
             {
-                LogActionSelection(actionSelected: true, actionInvoked: false, handled: context.IsHandled);
-
                 throw new InvalidOperationException(
                     Resources.FormatActionInvokerFactory_CouldNotCreateInvoker(
                         actionDescriptor.DisplayName));
             }
 
-            await invoker.InvokeAsync();
+            var controllerActionDescriptor = actionDescriptor as ControllerActionDescriptor;
+            var controllerName = string.Empty;
+            if (controllerActionDescriptor != null)
+            {
+                controllerName = controllerActionDescriptor.ControllerTypeInfo.FullName;
+            }
```

Comment:
You can get fancy with `?.` if you want to earn some PranavPoints to earn cool prizes!